django/db/migrations/state.py

```diff
@@ -180,10 +179,10 @@ def __init__(self, real_apps, models, ignore_swappable=False):
                 # (only for the first issue). Error message should look like:
                 # "ValueError: Lookup failed for model referenced by
                 # field migrations.Book.author: migrations.Author"
-                msg = "Lookup failed for model referenced by field {field}: {model[0]}.{model[1]}"
-                raise ValueError(msg.format(field=operations[0][1], model=lookup_model))
+                msg = "Lookup failed for model: {model[0]}.{model[1]}"
+                raise ValueError(msg.format(model=lookup_model))
             else:
-                do_pending_lookups(model)
+                self.do_pending_operations(model)
```

Comment:
This might be one for @andrewgodwin: this branch is never hit in the test suite. Whenever this for block is entered, ``ValueError`` is raised. The comment says that fields are sometimes added after construction – can we identify when that might happen and add a test, or else rewrite this block to just raise the ``ValueError`` if ``_pending_operations`` isn't empty?